python/ray/data/_internal/execution/interfaces/op_runtime_metrics.py

```diff
@@ -425,6 +425,16 @@ class OpRuntimeMetrics(metaclass=OpRuntimesMetricsMeta):
         description="Byte size of blocks in the external outqueue",
         metrics_group=MetricsGroup.OUTPUTS,
     )
+    num_prefetched_blocks: int = metric_field(
+        default=0,
+        description="Number of outstanding prefetched blocks in the iterator",
+        metrics_group=MetricsGroup.OUTPUTS,
+    )
+    num_prefetched_bytes: int = metric_field(
+        default=0,
+        description="Byte size of outstanding prefetched blocks in the iterator",
+        metrics_group=MetricsGroup.OUTPUTS,
+    )
```

Comment:
nit, these 2 metrics only make sense to the last operator. 
it'd be cleaner to attach them to the whole dataset, instead of individual ops.